File: java_integration/app/src/main/java/com/firewok/example/java_integration/shopping/shoppingcart/ShoppingCartFragment.java
package com.firewok.example.java_integration.shopping.shoppingcart;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import com.firework.imageloading.ImageLoader;
import com.firework.imageloading.glide.GlideImageLoaderFactory;
import com.firework.sdk.FireworkSdk;
import com.firewok.example.java_integration.databinding.FragmentCheckoutBinding;
import java.util.ArrayList;
import java.util.List;

public class ShoppingCartFragment extends Fragment {
    private FragmentCheckoutBinding binding;
    private ImageLoader imageLoader;
    private ShoppingCartAdapter checkoutAdapter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentCheckoutBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        
        imageLoader = GlideImageLoaderFactory.INSTANCE.createInstance(requireContext());
        checkoutAdapter = new ShoppingCartAdapter(imageLoader);
        
        binding.rvList.setAdapter(checkoutAdapter);
        binding.checkoutBtn.setOnClickListener(v -> {
            ShoppingCartRepository.clearOrders();
            FireworkSdk.INSTANCE.getShopping().setNumberOfItemsInCart(0);
            FireworkSdk.INSTANCE.getShopping().dismiss();
            
            if (getActivity() instanceof ShoppingCartListener) {
                ((ShoppingCartListener) getActivity()).onCheckout();
            }
            
            Toast.makeText(requireContext(), "Item purchased", Toast.LENGTH_SHORT).show();
        });
    }

    @Override
    public void onResume() {
        super.onResume();
        
        List<ShoppingCartRepository.Order> orders = ShoppingCartRepository.getInstance().getOrders();
        List<ProductUnitItem> items = new ArrayList<>();
        
        for (ShoppingCartRepository.Order order : orders) {
            String title = order.getProduct() != null ? order.getProduct().getName() : "";
            String subTitle = "";
            String iconUrl = "";
            
            if (order.getUnit() != null) {
                if (order.getUnit().getPrice() != null) {
                    subTitle = order.getUnit().getPrice().getAmount() + " " + order.getUnit().getPrice().getCurrencyCode();
                }
                if (order.getUnit().getImage() != null) {
                    iconUrl = order.getUnit().getImage().getUrl();
                }
            }
            
            items.add(new ProductUnitItem(title, subTitle, iconUrl));
        }
        
        checkoutAdapter.setData(items);
    }

    @Override
    public void onDestroyView() {
        binding = null;
        super.onDestroyView();
    }
} 

File: shopping/app/src/main/java/tv/fw/example/shoppingcart/ShoppingCartFragment.kt
package tv.fw.example.shoppingcart

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import tv.fw.example.databinding.FragmentCheckoutBinding
import tv.fw.fireworksdk.FireworkSdk
import tv.fw.imageloading.glide.GlideImageLoaderFactory

class ShoppingCartFragment : Fragment() {
    private var _binding: FragmentCheckoutBinding? = null
    private val binding: FragmentCheckoutBinding get() = _binding!!

    private val imageLoader by lazy {
        GlideImageLoaderFactory.createInstance()
    }

    private val checkoutAdapter by lazy {
        ShoppingCartAdapter(imageLoader)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View {
        _binding = FragmentCheckoutBinding.inflate(inflater)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.rvList.adapter = checkoutAdapter
        binding.checkoutBtn.setOnClickListener {
            ShoppingCartRepository.clearOrders()
            FireworkSdk.shopping.numberOfItemsInCart = 0
            FireworkSdk.shopping.dismiss()
            (activity as? ShoppingCartListener)?.onCheckout()
            Toast.makeText(requireContext(), "Item purchased", Toast.LENGTH_SHORT).show()
        }
    }

    override fun onResume() {
        super.onResume()
        checkoutAdapter.setData(
            ShoppingCartRepository.getOrders().map { order ->
                ProductUnitItem(
                    title = order.product?.name ?: "",
                    subTitle = "${order.unit?.price?.amount} ${order.unit?.price?.currencyCode}",
                    iconUrl = order.unit?.image?.url ?: "",
                )
            },
        )
    }

    override fun onDestroyView() {
        _binding = null
        super.onDestroyView()
    }
}
